### PR TITLE
Use AnySceneImporter for loading arbitrary formats

### PR DESCRIPTION
**Pull request by mosra**
## Motivation and Context

As mentioned in #54, current code had hardcoded only support for glTF files and in order to import other formats, users were forced to convert them to glTF. That's making users' life unnecessarily hard -- we're using Assimp anyway, so why not use it for general data as well. This PR enables effortless import of "any" formats using [AnySceneImporter](https://doc.magnum.graphics/magnum/classMagnum_1_1Trade_1_1AnySceneImporter.html) -- FBX, Blend, ..., basically anything it can recognize (and load a plugin for).

On the buildsystem side it's just re-enabling AnySceneImporter and AssimpImporter that were commented out right now, plus making the Assimp finding & linking a bit more robust.

Some formats (such as OBJ or PLY) don't provide scene hierarchy, in that case the code takes a wild guess and tries to import the first found mesh with the first found material.

## How Has This Been Tested

Built and tested with both system and embedded Assimp.

Loading PLY, OBJ and various other formats in the viewer utility, explicitly verified also with files that have no meshes and/or no materials, however I'm a bit unsure about the bounds checks -- let me know if these are okay.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/esp/assets/ResourceManager.cpp [hunk 1]
```diff
@@ -475,8 +484,20 @@ bool ResourceManager::createScene(Importer& importer,
     for (auto objectID : sceneData->children3D()) {
       createObject(importer, info, metaData, sceneNode, drawables, objectID);
     }
+  } else if (importer.mesh3DCount() && meshes_[metaData.meshIndex.first]) {
+    // no default scene --- standalone OBJ/PLY files, for example
+
+    // create scene parent node with transformation aligning to global frame
+    auto& sceneNode = parent.createChild();
+    const quatf transform =
+        quatf::FromTwoVectors(info.frame.front(), geo::ESP_FRONT);
```
Comment (msavva) [inline src/esp/assets/ResourceManager.cpp]: This should probably be `quatf::FromTwoVectors(info.frame.up(), geo::ESP_UP);` instead. We explicitly define the up vector for most of our assets, but leave front to the default in many cases.

Comment (mosra) [reply src/esp/assets/ResourceManager.cpp]: This is verbatim copied from the branch above (see line 480), should I change both?

**Comment by mosra** [comment]
> in the changes I've been making in #53 I've actually separated out assimp as a dependency as it was only used for building the datatool binary

@msavva actually, checking the SceneLoader.cpp, I propose rewriting the code to use AnySceneImporter as well instead of Assimp directly -- it only does manual mesh data import + fixing up the gravity orientation, right? Then we're depending on Assimp only as an (optional) plugin.

> OTOH, maybe we ought to try out building assimp on emscripten anyways. :smile:

I'm not saying it's impossible, but once it's just a plugin it's easy to replace it with dedicated importers for just the formats we want to support on the web (for example PLY, OBJ and glTF), saving *several megabytes* (the native binary is 8.5 MB on my machine, for reference). Assimp has a lot of functionality we don't need at all, it's not just import but also export and several complex tools for mesh processing.

**Comment by msavva** [comment]
> > in the changes I've been making in #53 I've actually separated out assimp as a dependency as it was only used for building the datatool binary
> 
> @msavva actually, checking the SceneLoader.cpp, I propose rewriting the code to use AnySceneImporter as well instead of Assimp directly -- it only does manual mesh data import + fixing up the gravity orientation, right? Then we're depending on Assimp only as an (optional) plugin.
> 
> > OTOH, maybe we ought to try out building assimp on emscripten anyways. 😄
> 
> I'm not saying it's impossible, but once it's just a plugin it's easy to replace it with dedicated importers for just the formats we want to support on the web (for example PLY, OBJ and glTF), saving _several megabytes_ (the native binary is 8.5 MB on my machine, for reference). Assimp has a lot of functionality we don't need at all, it's not just import but also export and several complex tools for mesh processing.

Sounds good on both counts!

File: src/esp/assets/ResourceManager.cpp (hunk 1 above)
Comment (msavva) [reply src/esp/assets/ResourceManager.cpp]: Ah! Let's leave the current behavior in place for both branches but add a "TODO ought to align with ESP_UP" above them.

Comment (mosra) [reply src/esp/assets/ResourceManager.cpp]: #61 by @erikwijmans fixes that, so I propose merging that one first -- and then I update the above to be the same as done in #61.

**Comment by mathfac** [comment]
If this PR will be rebased on the latest master we will get proper CI testing.

**Comment by mosra** [comment]
Rebased on master, adapted to the change from #61.

**Comment by msavva** [comment]
@mosra now that this PR is rebased and passed through CI, it should be ready to land on master and take advantage of the additional format support. 🙂

Please squash-and-merge into the master branch.

**Pull request merged**